apps/web/src/middleware.ts

```diff
@@ -1,15 +1,11 @@
-import { shouldGeoBlock } from '@pancakeswap/utils/geoBlock'
-import { NextRequest, NextResponse } from 'next/server'
+// middleware.ts
+import { withABHeaders } from 'middlewares/ab-test-middleware'
+import { withGeoBlock } from 'middlewares/geo-block-middleware'
+import { withUserIpHeaders } from 'middlewares/ip-address-middleware'
+import { stackMiddlewares } from 'middlewares/stack-middleware'
 
-export async function middleware(req: NextRequest) {
-  const res = NextResponse.next()
-
-  if (shouldGeoBlock(req.geo)) {
-    return NextResponse.redirect(new URL('/451', req.url))
-  }
-
-  return res
-}
+const middlewares = [withGeoBlock, withUserIpHeaders, withABHeaders]
+export default stackMiddlewares(middlewares)
```

Comment:
Only define a new variable when you have to. Another naming suggestion
```suggestion
export const middleware = useMiddlewares([withGeoBlock, withUserIpHeaders, withABHeaders])
```